app/gui2/src/components/GraphEditor.vue

```diff
@@ -88,6 +91,18 @@ function moveNode(id: ExprId, delta: Vec2) {
   const newPosition = node.position.addScaled(delta, 1 / navigator.scale)
   graphStore.setNodePosition(id, newPosition)
 }
+
+// === Populate CSS variables for group colors. ===
+
+watch([suggestionDb.groups, viewportNode], ([groups, root]) => {
+  if (root) {
+    for (let group of groups) {
+      const name = group.name.replace(/\s/, '-')
+      let color = group.color ?? colorFromString(name)
+      root.style.setProperty(`--group-color-${name}`, color)
+    }
+  }
```

Comment:
This should be done as part of `:style` binding in the template. That way you don't have to use `viewportNode` ref for this purpose at all. Direct DOM manipulation should rather be avoided.

Reply:
Fixed